Add single argument form of diff.namespace

sullyj3 opened the pull request:
## Overview

You can now do

`.> diff.namespace before`

to diff `before` with the current namespace.

Closes #1531

## Implementation notes

Modifies the appropriate `InputPattern`

## Interesting/controversial decisions

None, fairly straightforward change

## Test coverage

Added a line to `diff.md` to check for same output as two argument form.

## Loose ends

none.

sullyj3 committed:
Add single argument form of diff.namespace

aryairani commented on parser-typechecker/src/Unison/CommandLine/InputPatterns.hs:
```diff
@@ -926,19 +926,26 @@ diffNamespace :: InputPattern
       )
+    , ( "`diff.namespace before`"
+      , P.wrap
+        "shows how the namespace `before` differs from the current namespace"
```
for symmetry with the previous example, let's say "shows how the current namespace differs from the namespace `before`"

sullyj3 replied:
You're right, that makes more sense

sullyj3 committed:
Merge branch 'trunk' into diffWithCurrentNamespace

sullyj3 committed:
Reword diff.namespace help text for consistency

sullyj3 committed:
Add a failing test for single argument form of diff.namespace

The output for

```
.b2> diff.namespace .b1
```

should be the same as the output for

```
.> diff.namespace b1 b2
```

What the heck?

sullyj3 committed:
Add single '.' to diff.namespace test that I just spent too long debugging

sullyj3 commented:
Ok I'm happy with this, let me know if I need to add any more tests/other forms of testing.

pchiusano commented:
Great, thanks so much for knocking this out. I have definitely missed this.

sullyj3 commented:
No problem ☺️
